core/src/main/java/org/apache/calcite/sql/dialect/HiveSqlDialect.java

```diff
@@ -149,6 +127,22 @@ private void unparseTrim(SqlWriter writer, SqlCall call, int leftPrec,
   @Override public boolean supportsCharSet() {
     return false;
   }
+
+  @Override public SqlNode getCastSpec(final RelDataType type) {
+    if (type instanceof BasicSqlType) {
+      switch (type.getSqlTypeName()) {
+      case INTEGER:
+        final SqlBasicTypeNameSpec typeNameSpec = new SqlBasicTypeNameSpec(
+            INT,
+            type.getPrecision(),
+            type.getScale(),
```

Comment:
Use `SqlUserDefinedTypeNameSpec` and pass in the name you want to use instead, there is no need to add a new `SqlTypeName`.

Reply:
Thx